src/main/webapp/app/shared/utils/Utils.tsx

```diff
@@ -687,7 +687,13 @@ export function getAlterationName(
     typeof alteration === 'string' ? alteration : alteration.name;
   const isCategoricalAlt = isCategoricalAlteration(alt);
   const hasExclusionArm = alt.includes('{');
-  if (!showNameDiff || alt === name || isCategoricalAlt || hasExclusionArm) {
+  if (
+    !showNameDiff ||
+    alt === name ||
+    isCategoricalAlt ||
+    hasExclusionArm ||
+    name.includes(alt)
```

Comment:
What does this condition do?

Reply:
To deal with when alteration is `Promoter` and name is `Promoter Mutation`, then we should `Promoter Mutation` only.